### PR TITLE
Chore/restructure through abstraction

### PR DESCRIPTION
**Pull request by Pavel-Durov**
I put a "test" on it (although I had to restructure the project quite a bit).
let me know what you think!

Screenshot 2022-11-28 at 14 34 19

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,4 +50,7 @@ init-sealedsecrets:
 init-vault:
 	kubectl exec --stdin=true --tty=true vault-0 -- /bin/sh
 
-init-dev:  init-stack init-secretmanager init-sealedsecrets
\ No newline at end of file
+init-dev:  init-stack init-secretmanager init-sealedsecrets
+
+buildmocks:
+	mockery --all --dir "./pkg/"
```
Comment (Pavel-Durov) [inline Makefile:56]: generating mocks for all interfaces in /pkg path

File: cmd/kubeseal-convert/secretsmanager_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	return buf.String(), err
+}
+
+func TestGetResult(t *testing.T) {
+	rootCmd.AddCommand(secretsmanagerCmd)
+
+	// mock kubeseal
+	mockKubeSeal := mocks.NewKubeSeal(t)
+	mockKubeSeal.On("BuildSecretFile", mock.Anything).Return()
+	KubeSeal = mockKubeSeal
```
Comment (Pavel-Durov) [inline cmd/kubeseal-convert/secretsmanager_test.go:35]: poor man's solution for dependency injection ;)

File: cmd/kubeseal-convert/root.go [hunk 3]
```diff
@@ -21,6 +23,8 @@ var (
 
 		},
 	}
+	KubeSeal       = kubesealconvert.New()
```
Comment (Pavel-Durov) [inline cmd/kubeseal-convert/root.go:26]: poor man's solution for dependency management but maybe its good enough

File: pkg/kubeseal-convert/handlers/kubesealconvert/kubeseal.go [hunk 4]
```diff
@@ -30,7 +40,7 @@ func Kubeseal(secret string) {
 		Stdout: os.Stdout,
 		Stderr: os.Stdout,
 	}
-
+	var cmdBuffer bytes.Buffer
```
Comment (Pavel-Durov) [inline pkg/kubeseal-convert/handlers/kubesealconvert/kubeseal.go:43]: moved it inside the function just so we wont have race conditions if called concurrently

File: cmd/kubeseal-convert/secretsmanager.go [hunk 5]
```diff
@@ -11,11 +11,14 @@ var secretsmanagerCmd = &cobra.Command{
 	Short:   "Convert AWS Secrets-Manager secrets",
 	Args:    cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		secretData := kubesealconvert.Secretsmanager(args[0], cmd)
-
-		// Build Sealed Secrets using AWS Secrets Manager data
-		kubesealconvert.BuildSecretFile(secretData)
-
+		secretVal := domain.SecretValues{
```
Comment (Pavel-Durov) [inline cmd/kubeseal-convert/secretsmanager.go:14]: this change keeps all the "cli" "cobra" logic in the cmd  package and not vie versa

File: pkg/kubeseal-convert/handlers/kubesealconvert/kubeseal.go (hunk 4 above)
Comment (EladLeev) [reply pkg/kubeseal-convert/handlers/kubesealconvert/kubeseal.go:43]: Nice catch, thanks

**Review by EladLeev — APPROVED** [review]
Super elegant, thanks! 🎉

**Pull request merged**